packages/pangea-sdk/pangea/services/authz.py

```diff
@@ -0,0 +1,344 @@
+
+
+class TupleDeleteRequest(APIRequestModel):
+    tuples: List[Tuple]
+
+
+class TupleDeleteResult(PangeaResponseResult):
+    pass
+
+
+class CheckRequest(APIRequestModel):
+    resource: Resource
+    action: str
+    subject: Subject
+    debug: Optional[bool] = None
+    attributes: Optional[Dict[str, Any]] = None
+
+
+class DebugPath(APIResponseModel):
+    namespace: str
+    id: str
+    action: Optional[str]
+
+
+class Debug(APIResponseModel):
+    path: List[DebugPath]
+
+
+class CheckResult(PangeaResponseResult):
+    schema_id: str
+    schema_version: int
+    depth: int
+    allowed: bool
+    debug: Optional[Debug] = None
+
+
+class ListResourcesRequest(APIRequestModel):
+    namespace: str
+    action: str
+    subject: Subject
+
+
+class ListResourcesResult(PangeaResponseResult):
+    ids: List[str]
+
+
+class ListSubjectsRequest(APIRequestModel):
+    resource: Resource
+    action: str
+
+
+class ListSubjectsResult(PangeaResponseResult):
+    subjects: List[Subject]
+
+
+class AuthZ(ServiceBase):
+    """AuthZ service client.
+
+    Provides methods to interact with the Pangea AuthZ Service.
+    Documentation for the AuthZ Service API can be found at:
+    [https://pangea.cloud/docs/api/authz](https://pangea.cloud/docs/api/authz)
+
+    The following information is needed:
+        PANGEA_TOKEN - service token which can be found on the Pangea User
+            Console at [https://console.pangea.cloud/project/tokens](https://console.pangea.cloud/project/tokens)
+
+    Examples:
+        import os
+        from pangea.config import PangeaConfig
+        from pangea.services import AuthZ
+
+        PANGEA_TOKEN = os.getenv("PANGEA_AUTHZ_TOKEN")
+
+        authz_config = PangeaConfig(domain="aws.us.pangea.cloud")
+
+        # Setup Pangea AuthZ service client
+        authz = AuthZ(token=PANGEA_TOKEN, config=authz_config)
+    """
+
+    service_name = "authz"
+
+    def __init__(self, token, config=None, logger_name="pangea", config_id: Optional[str] = None):
+        super().__init__(token, config, logger_name, config_id=config_id)
+
+    def tuple_create(self, tuples: List[Tuple]) -> PangeaResponse[TupleCreateResult]:
+        """Create tuples.
+
+        Create tuples in the AuthZ Service. The request will fail if there is no schema
+        or the tuples do not validate against the schema.
+
+        Args:
+            tuples (List[Tuple]): List of tuples to be created.
+
+        Raises:
+            PangeaAPIException: If an API Error happens.
+
+        Returns:
+            Pangea Response with empty result.
+            Available response fields can be found in our
+            [API Documentation](https://pangea.cloud/docs/api/authz#tuple-create).
+
+        Examples:
+            # TODO: complete
+        """
+
+        input_data = TupleCreateRequest(tuples=tuples)
+        return self.request.post("v1beta/tuple/create", TupleCreateResult, data=input_data.dict(exclude_none=True))
+
+    def tuple_list(
```

Comment:
similar comment from the javascript sdk, but should tuple be its own class, so that this is more tuple.list